pkg/app/app.go

```diff
@@ -161,7 +162,7 @@ func (app *App) RegisterService(srv interface{}) error {
 	if rSrv, ok := srv.(common.Runnable); ok {
 		app.services = append(app.services, reflect.ValueOf(rSrv))
 	} else {
-		errMessage := "registered service is not a runnable"
+		errMessage := fmt.Sprintf("registered service is not a runnable: %T", srv)
```

Comment:
parameters are not added in the error message, they are added in the typed data of the log. So here what you want is:
```
errMessage := "registered service is not a runnable"
app.logger.With("service", srv).Error(errMessage)
return errors.ConfigError(errMessage)
```
That is called structured logging

Reply:
You're right, I could have done that.
Some [other error string contradict that](https://github.com/ConsenSys/quorum-key-manager/blob/50e060562dca6060270456dfb1752459621c8d67/pkg/app/app.go#L102), so I just inspired myself from what was there. Also, nothing is documented for the logger interface, which makes more digging. We could figure out kind of easily that it's zap behind, but it's more and more reference to dig.
Also, I added that because it helped me debug the app. I think it should be part of another PR.
I'm gonna remove it and make another PR.

Reply:
referenced here: https://github.com/ConsenSys/quorum-key-manager/issues/239